src/Hyperion/Analysis.hs

```diff
@@ -64,3 +70,7 @@ analyze ident samp = Report
         Sum
         (foldMapOf (measurements.each.batchSize.to realToFrac))
         samp
+    ([slope,yIntercept],determinationCoefficient) = (\(v,r) -> (UV.toList v,r)) $ olsRegress [batchSizesVect] durationsVect 
```

Comment:
It looks to me that it should be `olsRegress [durationVect] batchSizeVect`. Am I wrong?

Reply:
```suggestion
    ([slope,yIntercept],determinationCoefficient) = over _1 UV.toList $ olsRegress [batchSizesVect] durationsVect 
```

A bit of lens-fu here.

Reply:
I had to wrap my head around it a bit, but I think the order of arguments in the call to `olsRegress` is correct: this function is designed to infer a multivariate function of the form y = a1 * x1 + a2 * x2 + ... + an * xn + b, where the unknown is (a1,a2, ... an, b). So in our case, **x** is potentially of high dimension, hence the vector.
Our working hypothesis, if I'm not mistaken, is the following: batchDuration = benchDuration * batchSize + measureTime.
batchDuration unifies with _y_ in my previous example, and batchSize unifies with _x_ ; I think the correct invocation is indeed `olsRegress [batchSizesVect] durationsVect`